src/ekat/kokkos/ekat_subview_utils.hpp

```diff
@@ -316,6 +316,191 @@ subview_1(const ViewLR<ST******,Props...>& v,
       v.impl_track(),vm);
 }
 
+// ================ Multi-sliced Subviews ======================= //
+// e.g., instead of a single-entry slice like v(:, 42, :), we slice over a range
+// of values, as in v(:, 27:42, :)
+// this means that the subview has the same rank as the "parent" view
+
+// --- Rank1 multi-slice --- //
+template <typename ST, typename... Props>
+KOKKOS_INLINE_FUNCTION
+Unmanaged<ViewLS<ST*, Props...>>
+subview(const ViewLR<ST*, Props...>& v,
+        const Kokkos::pair<int, int> &kp0,
+        const int idim) {
+  assert(v.data() != nullptr);
+  assert(idim >= 0 && idim == v.rank);
+  // NOTE: the final comparison is originally int <= long unsigned int
+  // the cast silences a warning, but may be unnecessary
+  assert(kp0.first >= 0 && kp0.first < kp0.second
+         && (unsigned int)kp0.second <= v.extent(idim));
+  if (idim == 0) {
+    return Unmanaged<ViewLS<ST*,Props...>>(Kokkos::subview(v, kp0, Kokkos::ALL));
+  } else {
+    // FIXME: better way to do this? necessary?
+    assert(false);
+  }
+}
+
+// --- Rank2 multi-slice --- //
+template <typename ST, typename... Props>
+KOKKOS_INLINE_FUNCTION
+Unmanaged<ViewLS<ST**, Props...>>
+subview(const ViewLR<ST**, Props...>& v,
+        const Kokkos::pair<int, int> &kp0,
+        const int idim) {
+  assert(v.data() != nullptr);
+  assert(idim >= 0 && idim <= v.rank);
+  // NOTE: the final comparison is originally int <= long unsigned int
+  // the cast silences a warning, but may be unnecessary
+  assert(kp0.first >= 0 && kp0.first < kp0.second
+         && (unsigned int)kp0.second <= v.extent(idim));
+  if (idim == 0) {
+    return Unmanaged<ViewLS<ST**,Props...>>(Kokkos::subview(v, kp0, Kokkos::ALL));
+  } else if (idim == 1) {
+    return Unmanaged<ViewLS<ST**,Props...>>(Kokkos::subview(v, Kokkos::ALL, kp0));
+  } else {
```

Comment:
I don't think we want this exact pattern since it will lead to warnings (code branch without a return statement). You've already asserted earlier that `idim >= 0 && idim <= v.rank`, so we should never hit the else. I would do it this way:
```
if (idim == 0) {
    return Unmanaged<ViewLS<ST**,Props...>>(Kokkos::subview(v, kp0, Kokkos::ALL));
} else {
    assert(idim == 1); // probably redundant but doesn't hurt anything
    return Unmanaged<ViewLS<ST**,Props...>>(Kokkos::subview(v, Kokkos::ALL, kp0));
}
```

Reply:
I like that--thanks for the suggestion!